src/goto-symex/symex_main.cpp

```diff
@@ -67,20 +85,47 @@ void symex_transition(
     // 1. the transition from state.source.pc to "to" is not a backwards goto
     // or
     // 2. we are arriving from an outer loop
+
+    // TODO: This should all be replaced by natural loop analysis.
```

Comment:
In a future PR?

Reply:
Yeah, full natural loop analysis isn't in a position to use fully yet, why the current implementation is a mash-up of both approaches and also hidden behind the complexity flag.